core/src/main/java/com/airbnb/aerosolve/core/models/FullRankLinearModel.java

```diff
@@ -0,0 +1,164 @@
+package com.airbnb.aerosolve.core.models;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.*;
+
+import com.airbnb.aerosolve.core.DebugScoreRecord;
+import com.airbnb.aerosolve.core.FeatureVector;
+import com.airbnb.aerosolve.core.LabelDictionaryEntry;
+import com.airbnb.aerosolve.core.ModelHeader;
+import com.airbnb.aerosolve.core.ModelRecord;
+import com.airbnb.aerosolve.core.MulticlassScoringResult;
+import com.airbnb.aerosolve.core.util.Util;
+import com.airbnb.aerosolve.core.util.FloatVector;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// A full rank linear model that supports multi-class classificaiton.
+// The class vector Y = W' X where X is the feature vector.
+public class FullRankLinearModel extends AbstractModel {
```

Comment:
it seems this is mainly for multi-class classification, why use the name `FullRankLinearModel`